text/0000-reseeding-rand.md

```diff
@@ -0,0 +1,682 @@
+- Start Date: 2013-01-23
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+
+Stabilise the `std::rand` module by focusing it on a smaller set of
+functionality, moving "fancy" functionality to a crates.io crate.
+
+# Motivation
+
+The `std::rand` module provides various facilities for working with
+random numbers and randomness in general. This is a surprisingly
+sensitive topic, with different use cases. `std` does not aim to
+completely satisfy everything, but rather aims to provide a minimal
+set of functionality that tries to be simple and straightforward
+without restricting people building on it, and this reasoning extends
+to `std::rand`.
+
+A major area of use for randomness is cryptography where the
```

Comment:
The people who know what they're doing (like DJB and OpenBSD) aren't making this recommendation. The OS random-number generator is way too slow for most use cases requiring a CSPRNG and having it as the only option just encourages people to write insecure software. The `arc4random` API on OpenBSD is such a good thing *because* it can be safely used for any purpose - including as a CSPRNG - and yet it is extremely fast (a good implementation of chacha20). It's not clear why putting all of your trust in the kernel random-number generator is a good idea, especially considering that the Linux one is hacked together by non-cryptographers using primitives designed for other purposes (it mostly uses components from hashing algorithms, not a stream cipher).